src/sentry/sentry_metrics/indexer/cache.py

```diff
@@ -140,9 +157,18 @@ def bulk_record(
             return cache_key_results
 
         db_record_key_results = self.indexer.bulk_record(use_case_id, db_record_keys.mapping)
+
+        # We will temporarily change it to write new cache keys for all results
+
+        # DB key results
         self.cache.set_many(
             db_record_key_results.get_mapped_key_strings_to_ints(), use_case_id.value
         )
+
+        self.cache.set_many_new(
```

Comment:
We should control this via an option. That way we can easily control when to enable/disable this during runtime

Reply:
Good point, will add.

Reply:
https://github.com/getsentry/sentry/pull/48020